core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -200,9 +241,11 @@ class Partition(val topicPartition: TopicPartition,
   // defined when this broker is leader for partition
   @volatile private var leaderEpochStartOffsetOpt: Option[Long] = None
   @volatile var leaderReplicaIdOpt: Option[Int] = None
-  @volatile var inSyncReplicaIds = Set.empty[Int]
+  @volatile var isrState: IsrState = CommittedIsr(Set.empty)
```

Comment:
I wonder if we should be exposing this. Would it be enough to have a `def inSyncReplicaIds = isrState.isr`? One thing we need to be a little careful of is the fact that we now have a volatile variable with multiple fields. So if you try to access two fields through the `isrState` reference, you could see inconsistent data.

Reply:
Yea i was thinking we should move the ISR to a separate public accessor. I'll change this